resources/opendal/README.md

```diff
@@ -0,0 +1,41 @@
+# Shuttle OpenDAL
+
+This plugin allows services to connect to [Apache OpenDAL™](https://github.com/apache/opendal). OpenDAL is a data access layer that allows users to easily and efficiently retrieve data from various storage services in a unified way.
+
+Users can connect OpenDAL to access data from a variety of storage services, including: s3, azblob, gcs, oss and [so on](https://opendal.apache.org/docs/rust/opendal/services/index.html).
+
+## Usage
+
+**IMPORTANT**: Currently Shuttle isn't able to provision a storage for you (yet). This means you will have to create the storage service first and setup the secrets accordingly.
+
+Add `shuttle-opendal` to the dependencies for your service by running `cargo add shuttle-opendal`.
+This resource will be provided by adding the `shuttle_opendal::Opendal` attribute to your Shuttle `main` decorated function.
+
+It returns a `opendal::OpenDAL` for you to connect the storage service.
+
+### Example
+
+In the case of an Axum server, your main function will look like this:
+
+```rust
+use opendal::Operator;
+use shuttle_axum::ShuttleAxum;
+
+#[shuttle_runtime::main]
+async fn app(
+    #[shuttle_opendal::Opendal(scheme="s3")]
+    storage: Operator,
+) -> ShuttleAxum {}
+```
+
+### Parameters
+
+| Parameter | Type  | Default    | Description                                      |
+|-----------|-------|------------|--------------------------------------------------|
+| scheme    | `str` | `"memory"` | The scheme of the storage service to connect to. |
+
+All secrets are loaded from your `Secret.toml` file. 
+
+For instance, when using `s3`, you can configure the scheme to `s3` and specify the secrets: `bucket`, `access_key_id`, and `secret_access_key`.
```

Comment:
This is a clever way to load the config vars! However, there might be more secrets in `Secrets.toml` that are not related to the OpenDAL resource. Will that be a problem, or will they simply be ignored when initializing the operator?

Reply:
Looks like that's the case.